cadquery/occ_impl/exporters/assembly.py

```diff
@@ -61,3 +72,52 @@ def exportCAF(assy: AssemblyProtocol, path: str) -> bool:
     app.Close(doc)
 
     return status == PCDM_StoreStatus.PCDM_SS_OK
+
+
+def _vtkRenderWindow(assy: AssemblyProtocol) -> vtkRenderWindow:
+    """
+    Convert an assembly to a vtkRenderWindow. Used by vtk based exporters.
+    """
+
+    renderer = vtkRenderer()
+    renderWindow = vtkRenderWindow()
+    renderWindow.AddRenderer(renderer)
+    toVTK(assy, renderer)
+
+    renderer.ResetCamera()
+    renderer.SetBackground(1, 1, 1)
+
+    return renderWindow
+
+
+def exportVTKJS(assy: AssemblyProtocol, path: str):
+    """
+    Export an assembly to a zipped vtkjs. NB: .zip extensions is added to path.
+    """
+
+    renderer = vtkRenderer()
+    renderWindow = vtkRenderWindow()
+    renderWindow.AddRenderer(renderer)
+    toVTK(assy, renderer)
+
+    renderer.ResetCamera()
+    renderer.SetBackground(1, 1, 1)
+
+    with TemporaryDirectory() as tmpdir:
+
+        exporter = vtkJSONSceneExporter()
+        exporter.SetFileName(tmpdir)
+        exporter.SetRenderWindow(renderWindow)
+        exporter.Write()
+        make_archive(path, "zip", tmpdir)
+
+
+def exportVRML(assy: AssemblyProtocol, path: str):
```

Comment:
Thanks, I will give this a try for the KiCAD generator soon.

Reply:
HTH